packet/packet.go

```diff
@@ -221,14 +281,34 @@ func (packet *Packet) GetIPv4() *IPv4Hdr {
 	return nil
 }
 
-// GetIPv4 ensures if EtherType is IPv4 and cast L3 poinetr to IPv4Hdr type.
+// GetIPv4CheckVLAN ensures if EtherType is IPv4 and cast L3 poinetr
+// to IPv4Hdr type. If EtherType has VLANNumber, then VLAN header
+// EtherType is checked instead.
+func (packet *Packet) GetIPv4CheckVLAN() *IPv4Hdr {
+	if packet.GetEtherType() == SwapBytesUint16(IPV4Number) {
+		return (*IPv4Hdr)(packet.L3)
+	}
+	return nil
+}
+
+// GetARP ensures if EtherType is IPv4 and cast L3 poinetr to IPv4Hdr type.
 func (packet *Packet) GetARP() *ARPHdr {
 	if packet.Ether.EtherType == SwapBytesUint16(ARPNumber) {
 		return (*ARPHdr)(packet.L3)
 	}
 	return nil
 }
 
+// GetARPCheckVLAN ensures if EtherType is IPv4 and cast L3 poinetr to
```

Comment:
"GetARPCheckVLAN ensures if EtherType is ARP and cast L3 pointer to ARPHdr type."

Reply:
and castS